BREAKING(Transition): renamed Transition to Animation and added docs for the animations

mnajdova opened the pull request:
This PR is adding docs about how the animations should be defined in the theme, and used together with the Transition component, as well as the animation property.
# Update 
As part of this PR, I am going to also rename the `Transition` component to `Animation`, and the `animationName` property to `name`, following the review comments.

Commit:
-added documentation for the way the animations are defined in the theme

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,82 @@ export default () => (
+      <NavLink to="components/provider">Provider</NavLink>.
+    </p>
+    <p>
+      This is done with the Provider's <code>theme</code> prop. The animations are then are applied
```
> The animations are then *are* applied
remove the second 'are'

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,82 @@ export default () => (
+    <p>
+      This is done with the Provider's <code>theme</code> prop. The animations are then are applied
+      based on their name by using the <NavLink to="components/transition">Transition</NavLink>{' '}
+      component, or the <code>animation</code> property available on all UI component. Here's how we
```
nit: *all UI component* - I would rather write something like *all Stardust components*

miroslavstastny reviewed (commented):
The first question which comes to my mind is *when I should use `Transition` and when `animation`? In the original animation PR, there was a clear description and comparison of the two. Would it make sense to add similar description here as well?

levithomason commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,82 @@ export default () => (
+      ].join('\n')}
+      render={() => (
+        <div>
+          <Transition animationName="spinner">
```
This strikes me as odd, "Transition animationName=..." Should we rename this "Transition name=..." or "Animation name=..."?

mnajdova replied:
Thanks for the suggestion, I will rename the `Transition` component to `Animation`, as that is the word we use for describing this behavior everywhere else. `Transition` was just picked up because of SUIR. Then we can safely rename the `animationName` prop to `name`. I will add this changes to this PR too.

levithomason commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,82 @@ export default () => (
 
+    <Header as="h2" content="Animations" />
+    <p>
+      Another important part of the theming in Stardust are the <code>animations</code>. You can
```
>Another important part of the theming in Stardust are the <code>animations</code>.

This line doesn't add any value to the user's goal. Suggest removing it.

levithomason commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,82 @@ export default () => (
+    <Header as="h2" content="Animations" />
+    <p>
+      Another important part of the theming in Stardust are the <code>animations</code>. You can
+      define the animations in a very similar way as you would define them using css, by providing a
```
Suggest:

>You define animations in Stardust in a very similar way to CSS, by providing keyframes and animation properties.

Then show a minal provider code snippet immediately following this.

Commit:
-renamed Transition to Animation, renamed animationName to name
-addressed comments for the docs

Commit:
-added Animation to the theme typings

Commit:
Merge branch 'master' into doc/animation-theme

Commit:
-updated changelog

miroslavstastny commented on docs/src/examples/components/Animation/Types/AnimationExample.tsx:
```diff
@@ -0,0 +1,10 @@
+import { Animation, Icon } from '@stardust-ui/react'
+
+const AnimationExample = () => (
+  <Animation name="spinner">
```
All the `Animation` examples are theme dependent. Would it make more sense to add `Provider` to the examples and explicitly define the animations there?
Not sure about that, would like to hear other opinions.

mnajdova replied:
I understand your point, but I don't think this is good idea, because we will have much code in the examples, for something that is documented in the Theming guide. How the animation are created will be document as part of the theming guidem and the default theme will have few animations that will be shown in the examples. Let's hear the opinion from the other folks too. @kuzhelov @Bugaa92 @alinais @levithomason @layershifter

layershifter replied:
I have a similar problem in #451, we definitely need to show theme specific stuff because examples can be broken when you will try to switch to another theme.

> I understand your point, but I don't think this is good idea, because we will have much code in the examples, for something that is documented in the Theming guide. 

This is a valid point, too.

mnajdova replied:
After offline discussion we decided to add the Provider with the theme. All examples as well as the Theming.tsx guide page, now contains a Provider where the animations are defined.

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,106 @@ export default () => (
+    <ExampleSnippet
+      value={[
+        `<Animation name="spinner" delay="2s" duration="1s"><Icon name="user" circular /></Animation>`,
+        `<Icon name="book" animation="spinner" delay="5s" duration="2s" circular/>`,
```
`animation` here should be an object:
```jsx
<Icon name="book" animation={{name: "spinner", delay:"5s", duration:"2s"}} circular/>
```

mnajdova replied:
Right, thanks for the catch!

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,106 @@ export default () => (
+    />
+
+    <p>
+      You can also override some of the defined <code>animation</code> properties, by providing
```
I would explicitly mention here that `keyframe` cannot be overriden and can be only defined in the `Provider`.

mnajdova replied:
Good point! Will be sure to mention it.

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,106 @@ export default () => (
+          <Animation name="spinner" delay="2s" duration="1s">
+            <Icon name="user" circular />
+          </Animation>
+          <Icon name="book" animation="spinner" delay="5s" duration="2s" circular />
```
`animation` -> object

miroslavstastny commented on docs/src/views/Theming.tsx:
```diff
@@ -208,6 +208,106 @@ export default () => (
+
+    <p>
+      The difference between using the Animation component versus the animation property is that,
+      the Animation component can be safely use for applying animations on{' '}
```
use -> used

Commit:
-addressed PR comments
-renamed Animation theme interface to AnimationProp, because of conflicts with the Animation component name

layershifter commented on docs/src/examples/components/Animation/Types/AnimationExampleDelay.tsx:
```diff
@@ -0,0 +1,14 @@
+
+const AnimationExampleDelay = () => (
+  <div>
+    {'This animation will start after 5 seconds'}
```
```suggestion
    This animation will start after 5 seconds
```

Are brackets necessary?

mnajdova replied:
Removed.

Commit:
-removed unnecessary brackets

Commit:
-added Provider with animations in the examples and theming guide page and removed it form the teams' theme

Commit:
Merge branch 'master' into doc/animation-theme

Commit:
Merge branch 'master' into doc/animation-theme

# Conflicts:
#	CHANGELOG.md
